manifests/certonly.pp

```diff
@@ -46,7 +57,7 @@
   Array[Stdlib::Unixpath]                   $webroot_paths        = [],
   String[1]                                 $letsencrypt_command  = $letsencrypt::command,
   Integer[2048]                             $key_size             = $letsencrypt::key_size,
-  Array[String[1]]                          $additional_args      = [],
+  Optional[Array[String[1]]]                $additional_args      = undef,
```

Comment:
Is that actually required? Best practice for arrays is to default to an empty array instead of making it undef.

Reply:
I agree. I only used undef because it is being inserted into an argument array later where `delete_undef_values()` clears it if it's empty, so there will only be 1 space between arguments.

Reply:
OK, so I recently learned delete_undef_values is actually pretty oldschool and `.filter` can do a much better job. It will let me filter out the empty arrays AND undef. So I'll be able to switch this back to an empty array.